File: java/ChatAPI/src/chat/api/ServerConnectionInfo.java
package chat.api;

public class ServerConnectionInfo {
    public String name;
    public String ipOrDomain;
    public int port;
    public String publicKey;

    public ServerConnectionInfo(String name, String ipOrDomain, int port, String publicKey) {
        this.name = name;
        this.ipOrDomain = ipOrDomain;
        this.port = port;
        this.publicKey = publicKey;
    }

    @Override
    public String toString(){
        return "\""+name+"\" "+ipOrDomain+':'+port;
    }
}

File: c#/ChatAPI/ServerConnectionInfo.cs
namespace ChatAPI;

/// contains all information needed to connect to a server
public class ServerConnectionInfo
{
    /// server name
    public string name;
    /// server address
    public string ipOrDomain;
    /// server port
    public int port;
    /// server's public RSA encryption key
    public string publicKey;

    public ServerConnectionInfo(string ipOrDomain, int port, string name, string publicKey)
    {
        this.ipOrDomain = ipOrDomain;
        this.port = port;
        this.name = name;
        this.publicKey = publicKey;
    }

    public override string ToString() => $"\"{name}\" {ipOrDomain}:{port}";
}